.github/workflows/checks.yml

```diff
@@ -0,0 +1,41 @@
+name: checks
+
+on: [push, pull_request]
```

Comment:
If you push to a branch in the same repository as the PR is opened, this will run the checks twice (see the result of this run). This should be changed so only one run is done in this case.

Reply:
I know this is a different repo, but see discussion with Adam Tyson. I (think) I got persuaded to his line of thinking. Unless you can think of a better way of doing it/a more convincing argument, https://github.com/brainglobe/cellfinder-core/pull/33

![image](https://user-images.githubusercontent.com/15052188/168554973-71a3e308-290c-447a-92e3-994cb0735834.png)

Reply:
I don't understand that - there's no advantage running exactly the same tests with exactly the same configuration twice.

Reply:
This is easily fixed by specifying that on the `push` event the tests are only run if the branch is `main`, e.g. see https://github.com/sunpy/sunpy/blob/216b964d69a99feea812570a4ac64e37e23d3da3/.github/workflows/ci.yml#L3-L7

Reply:
Which if you look here https://github.com/brainglobe/cellfinder-core/pull/33/files is exactly what I did.

What happens if you raise a PR and keep pushing to it, does that definitely run each time? Another point (not that I encourage it) if someone works on a branch and merges it in without a PR, only then will we know that `main` is "broken"? Just playing devil's advocate, but I think these were Adam's concerns.

So, anyway, do you agree what I originally did made sense?

Reply:
> What happens if you raise a PR and keep pushing to it, does that definitely run each time? 

Yes

> Another point (not that I encourage it) if someone works on a branch and merges it in without a PR, only then will we know that `main` is "broken"?

Yes, which is why it's always good practice to open a PR prior to merging code. 

> So, anyway, do you agree what I originally did made sense?

As it currently stands I don't think this PR makes sense, I think it needs to be changed so two lots of identical workflows aren't run when opening a PR from a branch in the same repository.

Reply:
> Yes, which is why it's always good practice to open a PR prior to merging code.

Yes totally agree. Personally I tend to wait to the work is done before I open, but doesn't matter either way